### PR TITLE
Cleanup some more clang warnings

### PR DESCRIPTION
**Pull request by adizaimi**
Down to about 530 warnings (from about 1400) after this.

### CODE REVIEW

File: bbinc/fwd_types.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+/*
+ * forward declared types
+ */
+
+#ifndef __fwd_types_h__
+#define __fwd_types_h__
+
+typedef struct bdb_state_tag bdb_state_type;
+typedef struct UnpackedRecord UnpackedRecord;
+typedef struct Vdbe Vdbe;
```
Comment (riverszhang89) [inline bbinc/fwd_types.h]: It feels wrong to place SQLite definitions in `bbinc` and have SQLite include the file.

Comment (adizaimi) [reply bbinc/fwd_types.h]: Yeah, maybe i'll move it to sqlite/

**Comment by adizaimi** [comment]
2017/11/11 18:26:02 [ERROR] udp_bind:bind: Address already in use
for both the failed tests above.

**Comment by adizaimi** [comment]
Thank you Mike.

**Comment by adizaimi** [comment]
> Aren’t these in db/sql.h
Yes, but i had to remove it from there because of the warning of duplicate definition. Moving them to one place seems to be the only solution.

**Comment by akshatsikarwar** [comment]
That's cool. Didn't realize you did that.

**Pull request merged**